modules/light-clients/07-tendermint/types/update.go

```diff
@@ -60,6 +60,34 @@ func (cs ClientState) CheckHeaderAndUpdateState(
 		return nil, nil, err
 	}
 
+	// Check the earliest consensus state to see if it is expired, if so then set the prune height
+	// so that we can delete consensus state and all associated metadata.
+	var (
+		pruneHeight exported.Height
+		pruneError  error
+	)
+	pruneCb := func(height exported.Height) bool {
```

Comment:
It is a little odd to me that we cannot prune consensus states for expired clients. I guess this can be fixed in a future improvment

Reply:
Yes this can be added in future PR